### PR TITLE
Fixed length postcode validation for Liechtenstein.

### PR DESCRIPTION
**Pull request by bogdanarizancu**
### Ticket
- [Valid four-digit Liechtenstein postcodes marked as invalid, stopping orders](https://app.asana.com/0/587433704548192/1199387326781845/f)

### CODE REVIEW

**Review by juanlopez4691 — APPROVED** [review]

File: src/WooCommerce.php [hunk 1]
```diff
@@ -909,7 +909,7 @@ public static function woocommerce_validate_postcode($valid, $postcode, $country
         break;
 
       case 'LI':
-        $valid = (bool) preg_match('@^(948[5-9])|(949[0-7])$@', $postcode);
+        $valid = (bool) preg_match('@^((948[5-9]{1})|(949[0-7]{1}))$@', $postcode);
```
Comment (sun) [inline src/WooCommerce.php:912]: `{1}` is the default of each character.
```suggestion
        $valid = (bool) preg_match('@^((948[5-9])|(949[0-7]))$@', $postcode);
```

**Review by sun — CHANGES_REQUESTED** [review]

File: src/WooCommerce.php (hunk 1 above)
Comment (bogdanarizancu) [reply src/WooCommerce.php]: It is added to limit to 1 character length, it otherwise allows 5 digit postcodes, I have already tested this.

Comment (sun) [reply src/WooCommerce.php]: I do not see a repeater instruction in the expression, so this does not match 5 digits.

See for yourself here: https://regex101.com/r/jIZPN7/1

File: src/WooCommerce.php [hunk 2]
```diff
@@ -909,7 +909,7 @@ public static function woocommerce_validate_postcode($valid, $postcode, $country
         break;
 
       case 'LI':
-        $valid = (bool) preg_match('@^(948[5-9])|(949[0-7])$@', $postcode);
+        $valid = (bool) preg_match('@^((948[5-9]{1})|(949[0-8]{1}))$@', $postcode);
```
Comment (sun) [inline src/WooCommerce.php:912]:
```suggestion
        $valid = (bool) preg_match('@^((948[5-9])|(949[0-8]))$@', $postcode);
```

**Review by sun — APPROVED** [review]
As discussed above

File: src/WooCommerce.php (hunk 2 above)
Comment (bogdanarizancu) [reply src/WooCommerce.php:912]: Now I see what you mean, thank you.

**Pull request merged**